compiler/repl-service/server/src/main/scala/com/digitalasset/daml/lf/ReplServiceMain.scala

```diff
@@ -177,6 +179,22 @@ object ReplService {
   private val compilerConfig = Compiler.Config.Default.copy(
     stacktracing = Compiler.FullStackTrace
   )
+  private val homePackageId: PackageId = PackageId.assertFromString("-homePackageId-")
+
+  private def moduleRefs(v: SValue): Set[ModuleName] = {
+    def moduleRefs(acc: Set[ModuleName], e: SExpr): Set[ModuleName] =
+      e match {
+        case SEVal(ref) =>
+          if (ref.packageId == homePackageId) {
+            acc + ref.modName
+          } else {
+            acc
+          }
+        case e => SExprIterable(e).foldLeft(acc)(moduleRefs)
+      }
+
+    SExprIterable(v).foldLeft(Set.empty[ModuleName])(moduleRefs)
+  }
```

Comment:
Can we move that to `com.daml.lf.speedy.SValue` and make the `SExprIterable` `speedy` private ?

Reply:
I kept this function here but moved the code under `com.daml.lf.speedy.repl.ReplServiceMain` and made the other stuff `private`.

Reply:
Thanks.